File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.js
function solution(n) {
    const three = n.toString(3);
    
    // console.log("3진법으로 변경", three);
    
    let reverse = '';
    
    for(let i = three.length - 1; i >= 0; i--){
        // console.log(`${i + 1}번째 문자`, three[i]);
        
        reverse += three[i];
    }
    
    // console.log("뒤집힌 삼진법", reverse);
    
    // console.log('10진법', parseInt(reverse, 3));
    
    return parseInt(reverse, 3)
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    
    if n < 3:
        return n 
    
    # 3진법으로 변환 
    trinary = []
    
    while True:
        trinary.append(str(n % 3))
        
        n = n // 3
        
        if n < 3:
            trinary.append(str(n))
            break
    
    # 10진수로 변환 
    return int("".join(trinary), 3)
    